File: 34.find-first-and-last-position-of-element-in-sorted-array.cpp
// @leet imports start
#include <bits/stdc++.h>
using namespace std;
// @leet imports end

// @leet start
class Solution
{
public:
    vector<int> searchRange(const vector<int>& nums, int target)
    {
        auto it = lower_bound(nums.begin(), nums.end(), target);
        if (it == nums.end() || *it > target)
            return {-1, -1};
        int left = it - nums.begin();
        auto it2 = upper_bound(it + 1, nums.end(), target);
        if (it2 == nums.end())
            return {left, static_cast<int>(nums.size() - 1)};
        return {left, static_cast<int>(it2 - nums.begin()) - 1};
    }
};
// @leet end

File: 34.find-first-and-last-position-of-element-in-sorted-array.py
# @leet imports start
from string import *
from re import *
from datetime import *
from collections import *
from heapq import *
from bisect import *
from copy import *
from math import *
from random import *
from statistics import *
from itertools import *
from functools import *
from operator import *
from io import *
from sys import *
from json import *
from builtins import *
import string
import re
import datetime
import collections
import heapq
import bisect
import copy
import math
import random
import statistics
import itertools
import functools
import operator
import io
import sys
import json
from typing import *
# @leet imports end

# @leet start
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        if not nums:
            return [-1, -1]

        # Find first occurrence
        lo, hi = 0, len(nums) - 1
        while lo < hi:
            mid = (lo + hi) // 2
            if nums[mid] < target:
                lo = mid + 1
            else:
                hi = mid

        if nums[lo] != target:
            return [-1, -1]

        left = lo

        # Find last occurrence
        lo, hi = left, len(nums) - 1
        while lo < hi:
            mid = (lo + hi + 1) // 2  # bias right
            if nums[mid] > target:
                hi = mid - 1
            else:
                lo = mid

        return [left, lo]
# @leet end
